### PR TITLE
feat(platform): remove comments by content

### PR DESCRIPTION
**Pull request by ZauberNerd**
This PR is in preparation for #6002 to keep the reviews smaller and more focussed.

When merging this PR we will be able to remove comments either by topic or by content:
```javascript
// by content
await platform.ensureCommentRemoval({
  number: 1,
  content: 'some content',
});

// by topic
await platform.ensureCommentRemoval({
  number: 1,
  topic: 'some topic',
});
```

All commits in this PR focus on one platform implementation and can be reviewed individually.

_Hint: By adding `?w=1` to the url some diffs become smaller, because white-space only changes won't be shown anymore._

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
If `topic` and `content` are mutually exclusive, ideally our API would ensure exactly one is selected, e.g. number, removalType, removalText (all of them mandatory). Is this feasible to do?

**Comment by ZauberNerd** [comment]
> If `topic` and `content` are mutually exclusive, ideally our API would ensure exactly one is selected, e.g. number, removalType, removalText (all of them mandatory). Is this feasible to do?

Yes, we can use TypeScript's function overloading for that: https://github.com/renovatebot/renovate/pull/6181/commits/ffdaac1390e7190e963bce47a58d3d882e3bd886

File: lib/platform/common.ts [hunk 1]
```diff
@@ -206,9 +206,14 @@ export interface Platform {
     branchName: string,
     context: string
   ): Promise<BranchStatus | null>;
-  ensureCommentRemoval(
-    ensureCommentRemoval: EnsureCommentRemovalConfig
-  ): Promise<void>;
+  ensureCommentRemoval(ensureCommentRemoval: {
```
Comment (viceice) [inline lib/platform/common.ts:209]: I would like to improve the types here, will send a commit later.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/platform/common.ts (hunk 1 above)
Comment (rarkins) [reply lib/platform/common.ts:209]: a commit to master after it's merged, or to this PR?

**Comment by viceice** [comment]
I can send it to this pr in a few minutes, working on it.

File: lib/platform/azure/index.ts [hunk 2]
```diff
@@ -572,29 +573,40 @@ export async function ensureComment({
...
-  if (issueNo) {
-    const azureApiGit = await azureApi.gitApi();
-    const threads = await azureApiGit.getThreads(config.repoId, issueNo);
-    let threadIdFound = null;
+  logger.debug(
+    `Ensuring comment "${topic || content}" in #${issueNo} is removed`
+  );
+  if (!issueNo) {
+    return;
+  }
```
Comment (viceice) [inline lib/platform/azure/index.ts:583]:
```suggestion
```
@rarkins I think this should never happen?

**Review by viceice — DISMISSED** [review]

File: lib/platform/common.ts (hunk 1 above)
Comment (viceice) [reply lib/platform/common.ts]: done

File: lib/platform/azure/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/platform/azure/index.ts:583]: right

Comment (viceice) [reply lib/platform/azure/index.ts:583]: ok, fixing it

**Review by viceice — APPROVED** [review]

**Pull request merged**